### PR TITLE
Clarify GITHUB_SHA for push event

### PR DESCRIPTION
**Pull request by dannycjones**
### Why:

Simple clarification on the value of `GITHUB_SHA` for `push` event. It's fairly obvious but with this change, there's no room for ambiguity.

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Just clarifying the content of `GITHUB_SHA`. First sentence under `GITHUB_SHA` is newly added.

>| Webhook event payload | Activity types | `GITHUB_SHA` | `GITHUB_REF` |
>| --------------------- | -------------- | ------------ | -------------|
>| [`push`](/webhooks-and-events/webhooks/webhook-events-and-payloads#push) | Not applicable | Commit pushed to the ref. When you delete a branch, the SHA in the workflow run (and its associated refs) reverts to the default branch of the repository. | Updated ref |

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by nguyenalex836** [comment]
@dannycjones Thanks so much for opening a PR! I'll get this triaged for review ✨

File: content/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows.md [hunk 1]
```diff
@@ -929,7 +929,7 @@ jobs:
 
 | Webhook event payload | Activity types | `GITHUB_SHA` | `GITHUB_REF` |
 | --------------------- | -------------- | ------------ | -------------|
-| [`push`](/webhooks-and-events/webhooks/webhook-events-and-payloads#push) | Not applicable | When you delete a branch, the SHA in the workflow run (and its associated refs) reverts to the default branch of the repository. | Updated ref |
+| [`push`](/webhooks-and-events/webhooks/webhook-events-and-payloads#push) | Not applicable | Commit pushed to the ref. When you delete a branch, the SHA in the workflow run (and its associated refs) reverts to the default branch of the repository. | Updated ref |
```
Comment (nguyenalex836) [inline content/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows.md:932]:
```suggestion
| [`push`](/webhooks-and-events/webhooks/webhook-events-and-payloads#push) | Not applicable | Tip commit pushed to the ref. When you delete a branch, the SHA in the workflow run (and its associated refs) reverts to the default branch of the repository. | Updated ref |
```

**Comment by nguyenalex836** [comment]
@dannycjones Thank you so much for your patience while our SME team reviewed! They found a small to the wording would offer increased clarity, which I have added as a suggestion 💛 

Once that suggestion has been applied, we will be happy to get this merged!

**Comment by dannycjones** [comment]
Thanks @nguyenalex836, I incorporated the feedback and also merged from `main`.

**Comment by nguyenalex836** [comment]
@dannycjones Thank you for applying that suggestion! I'll update the branch and get this merged once tests are passing 🍏

**Review by nguyenalex836 — APPROVED** [review]

**Pull request merged**